zkevm-circuits/src/state_circuit/constraint_builder.rs

```diff
@@ -19,17 +20,16 @@ pub struct Queries<F: Field> {
     pub rw_counter: MpiQueries<F, N_LIMBS_RW_COUNTER>,
     pub is_write: Expression<F>,
     pub tag: Expression<F>,
-    pub prev_tag: Expression<F>,
+    pub tag_bits: [Expression<F>; 4],
     pub id: MpiQueries<F, N_LIMBS_ID>,
-    pub is_id_unchanged: Expression<F>,
+    pub is_tag_and_id_unchanged: Expression<F>,
```

Comment:
we also need `is_tag_unchanged`  and `is_id_unchanged`  `is_field_tag_unchanged`,  I have use them in  PR #552

Reply:
we can add them later when they're needed.